app/controllers/users_controller.rb

```diff
@@ -6,7 +6,7 @@ class UsersController < ApplicationController
   include Foreman::TelemetryHelper
 
   skip_before_action :require_mail, :only => [:edit, :update, :logout, :stop_impersonation]
-  skip_before_action :require_login, :authorize, :session_expiry, :update_activity_time, :set_taxonomy, :set_gettext_locale_db, :only => [:login, :logout, :extlogout]
+  skip_before_action :require_login, :check_user_enabled, :authorize, :session_expiry, :update_activity_time, :set_taxonomy, :set_gettext_locale_db, :only => [:login, :logout, :extlogout]
```

Comment:
There's a few other occurrences of `skip_before_action :require_login` in various controllers, I think `check_user_enabled` should be skipped in those places as well since there won't be a user to check.

Reply:
if there is no user, the `:check_user_enabled` returns nil, as we use `&.` operator there. Maybe we don't need to skip it anywhere and add a comment there stating the behaviour as "If current user is present, checks if it is enabled"?

Reply:
I think it will be better to add (and it's pushed) `:check_user_enabled` to `skip...` statement where login is not required. If I remember correctly, this can broke a chain of validations in extreme case. Any opinions on that? @ezr-ondrej @tbrisker